### DISCUSSION TITLE
Traffic Split based on number of requests

### DISCUSSION THREAD

**Discussion by cham-d** [linkerd/linkerd2#7879] [Q&A] [answered]
Is there any way to split the traffic based on the number of requests?
For example: 
- Service1 with label = "myapp" will handle 200 requests all the time. (you must probably monitor how many requests service1 handles at a given time)
- Service2 with label = "myapp" will handle the rest of the requests (requests - 200).
Probably you can do this by recalculating and changing the percentage of traffic splits continuously or every x seconds, but I don't know if this kind of implementation will be so stable.

**Comment by kleimkuhler** [comment, marked as answer]
@cham-d There is not currently a way to use TrafficSplits based off of the number of requests. The closest you'll get is having a rough idea of the total number of requests the apex service is receiving and adjust the weights based off that.

There could be an operator that watches this and adjusts the weights dynamically—similar to how Linkerd's failover feature will be implemented—but this is not currently implemented to my knowledge.

**Reply by cham-d** [reply]
yeah, that's exactly the solution i thought, too. Thank you
